bigchaindb/upsert_validator/validator_election.py

```diff
@@ -2,264 +2,48 @@
-
-            if election.is_same_topology(current_validators, election.outputs):
-                total_votes = sum(current_validators.values())
-                if (votes_commited < (2/3)*total_votes) and \
-                   (votes_commited + votes_current >= (2/3)*total_votes):
-                    return election
-        return False
-
-    @classmethod
-    def get_validator_update(cls, bigchain, new_height, txns):
-        votes = {}
-        for txn in txns:
-            if not isinstance(txn, ValidatorElectionVote):
-                continue
-
-            election_id = txn.asset['id']
-            election_votes = votes.get(election_id, [])
-            election_votes.append(txn)
-            votes[election_id] = election_votes
-
-            election = cls.has_concluded(bigchain, election_id, election_votes, new_height)
-            # Once an election concludes any other conclusion for the same
-            # or any other election is invalidated
-            if election:
-                # The new validator set comes into effect from height = new_height+1
-                validator_updates = [election.asset['data']]
-                curr_validator_set = bigchain.get_validators(new_height)
-                updated_validator_set = new_validator_set(curr_validator_set,
-                                                          validator_updates)
-
-                updated_validator_set = [v for v in updated_validator_set if v['voting_power'] > 0]
-                bigchain.store_validator_set(new_height+1, updated_validator_set, election.id)
-
-                validator16 = encode_pk_to_base16(election.asset['data'])
-                return [encode_validator(validator16)]
-
-        return []
-
-    def get_validator_update_by_election_id(self, election_id, bigchain):
-        result = bigchain.get_validators_by_election_id(election_id)
-        return result
-
-    def get_status(self, bigchain):
-        concluded = self.get_validator_update_by_election_id(self.id, bigchain)
-        if concluded:
-            return self.CONCLUDED
-
-        latest_change = self.get_validator_change(bigchain)
-        latest_change_height = latest_change['height']
-        election_height = bigchain.get_block_containing_tx(self.id)[0]
+    def on_approval(cls, bigchain, election, new_height):
+        # The new validator set comes into effect from height = new_height+1
+        validator_updates = [election.asset['data']]
+        curr_validator_set = bigchain.get_validators(new_height)
+        updated_validator_set = new_validator_set(curr_validator_set,
+                                                  validator_updates)
 
-        if latest_change_height >= election_height:
-            return self.INCONCLUSIVE
-        else:
-            return self.ONGOING
+        updated_validator_set = [v for v in updated_validator_set if v['voting_power'] > 0]
+        bigchain.store_validator_set(new_height+1, updated_validator_set, election.id)
```

Comment:
As we decided in our discussion we will not store `election.id` in the validators collection instead store the fact that `election.id` concluded in the `elections` collections

Reply:
Done